common/djangoapps/student/views/dashboard.py

```diff
@@ -828,6 +828,11 @@ def student_dashboard(request):
             enr for enr in course_enrollments if entitlement.enrollment_course_run.course_id != enr.course_id
         ]
 
+    # Check whether we are showing all the courses or not
```

Comment:
`load_all_course = request and load_all_course in request.GET`, Same here, we should move it to the top of the view and pass to the functions instead of getting it multiple times.